x/sysproxy/sysproxy_darwin.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build darwin
+
+package sysproxy
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type ProxyType string
+
+const (
+	httpProxyType  ProxyType = "web"
+	httpsProxyType ProxyType = "secureweb"
+)
+
+func SetProxy(ip string, port string) error {
```

Comment:
@fortuna I agree it would be helpful to export different functions to set different types of proxies and perhaps one method that sets them all at once such as `SetProxyAll()`. This way developers can choose one method if they just want to set a specific type of proxy. 

- Also, I realized that HTTP proxy also allows for setting username/password for authentication. Do you thing we should expose that in the API? 

- Regarding socks proxy, I am researching on how to do it on Windows but if I can't get it working, we can just return `platform not supported` error for `SetSocksProxy` on Windows.

Reply:
@fortuna I did some further research on this. The table below summarizes system-wide support for various types of proxies on desktop platforms.

| Proxy Type | Windows | MacOS | Linux |
| --- | ----------- | ------ | ------ |
| HTTP | Yes (No Auth)<sup>1</sup> | Yes | Yes
| HTTPS | Yes (No Auth)<sup>1</sup> | Yes | Yes (No Auth)<sup>3</sup>
| FTP | No | No | Yes (No Auth)
| SOCKS | No | Yes (No Auth)<sup>2</sup> | Yes (No Auth)


1. Windows does not explicitly distinguish between HTTP and HTTPS proxy. Also, username/password authentication is not supported.

2. MacOS SOCKS does not seems to correctly support authentication even though it accepts credentials [[ref](https://discussions.apple.com/thread/255394737?sortBy=best)].

3. Username/Pass authentication is not currently supported for HTTPS proxy [[ref](https://gitlab.gnome.org/GNOME/gsettings-desktop-schemas/-/issues/42)].

Given the poor support for username/password authentication, do you think it makes sense to add authentication option to the `sysproxy` package?